### PR TITLE
Merge gpunload code to gpload and add test cases

### PR DESCRIPTION
**Pull request by 50wu**
gpunload the equivalent of gpload, but for the extract. It allows to
quickly extract a large set of data from one table. Globally, it performs
the following steps:

   1. From a configuration file build a writable external table
   2. Start a gpfdist process
   3. Run an INSERT SELECT from the input table to the built external table
   in order to extract the data from the input table into the output file.
   4. Stop the gpfdist process.

### CODE REVIEW

File: gpMgmt/bin/gpload.py [hunk 1]
```diff
@@ -1458,8 +1498,40 @@ def read_config(self):
             # like libpq, just inherit USER
             self.options.d = self.options.U
 
-        if self.getconfig('gpload:input:error_table', unicode, None):
-            self.control_file_error("ERROR_TABLE is not supported. Please use LOG_ERRORS instead.")
+        # Password
```
Comment (pivotal-mike) [inline gpMgmt/bin/gpload.py]: We seem to have dropped the check for error_table use that was originally here.  Should this be added back in?

Can we also be sure there's a test case for this?

File: gpMgmt/bin/gpload.py [hunk 2]
```diff
@@ -1458,8 +1498,40 @@ def read_config(self):
...
+                self.options.password = self.getconfig('password', unicode,
+                                                       None)
+            if self.options.password==None:
+                self.options.password = os.environ.get('PGPASSWORD')
+            if self.options.password==None:
+                self.readPgpass(os.environ.get('PGPASSFILE',
+                                os.environ.get('HOME','.')+'/.pgpass'))
+
+        # make decision to get parameter from goload yaml file or gpunload yaml file
+        self.choice = self.getconfig('gpload',dict,default=None)
```
Comment (pivotal-mike) [inline gpMgmt/bin/gpload.py]: Can we make this a little clearer?  It's to distinguish between gplaod and gpunload but it's being used in a true/false sense.  This make reading the code more difficult than it needs to be

Comment (50wu) [inline gpMgmt/bin/gpload.py]: Hi mike,

Thank you for the review and all your advice is good. As you said, the error_table code is dropped in gpdb5 but used in gpdb4, and I will clean them up in gpdb5. I will try to refactor the code to make it easy to read.

Again, thank you for taking the time to showcase your ideas and welcome to make more advice.

**Comment by Eulerizeit** [comment]
There seems to be quite a bit of discussion about what this feature does and what this feature should do on the #gpdb slack channel.  

Can we start a conversation about this feature on the gpdb-dev and gpdb-users email lists please?

**Comment by pivotal-mike** [comment]
@50wu Can we close this?

**Comment by pivotal-mike** [comment]
Hey Folks - this appears to be dead.  I'm going to close it for the moment.  If mistaken please re-open and let's resolve the outstanding conflicts and issues.

**Pull request closed without merging**